sync/worker_pool_test.go

```diff
@@ -121,3 +121,23 @@ func TestGoWithTimeout(t *testing.T) {
 
 	require.Equal(t, uint32(testWorkerPoolSize+1), count)
 }
+
+func TestFullWorkerPool(t *testing.T) {
```

Comment:
Not exactly sure what is this testing, if you remove the second Init() does the test still pass? Also should probably rename this test to something like `TestInitOnInitializedWorkerPool`

Reply:
yeah this test is a little odd because init does not return anything. I just wanted to show that this no longer hangs forever. Maybe the name change will make it clear.